app/models/concerns/effective_devise_user.rb

```diff
@@ -62,6 +62,28 @@ def effective_devise_user
     before_save(if: -> { persisted? && encrypted_password_changed? }) do
       assign_attributes(provider: nil, access_token: nil, refresh_token: nil, token_expires_at: nil)
     end
+
+    # Uniqueness validation of emails and alternate emails across all users
+    validate({ if: -> { respond_to?(:alternate_email) } }) do
+      records = self.class.where.not(id: self.id) # exclude self
+      email_duplicates = records.where("lower(email) = :email OR lower(alternate_email) = :email", email: email.to_s.strip.downcase)
+      alternate_email_duplicates = records.where("lower(email) = :alternate_email OR lower(alternate_email) = :alternate_email", alternate_email: alternate_email.to_s.strip.downcase)
+
+      # Check if a uniqueness validation was already performed before triggering the exists query
+      if !self.errors.added?(:email, 'has already been taken') && email_duplicates.exists?
+        self.errors.add(:email, 'has already been taken')
+      end
+
+      # Check if the alternate email is set before triggering the exists query
+      if alternate_email.present? && alternate_email_duplicates.exists?
```

Comment:
if `try(:alternate_email).present?`

We can't assume the user model will have an alternate email field. It's weird i know lol.